sky/execution.py

```diff
@@ -460,6 +460,19 @@ def exec(  # pylint: disable=redefined-builtin
     backend_utils.check_cluster_name_not_reserved(cluster_name,
                                                   operation_str='sky.exec')
 
+    if isinstance(entrypoint, sky.Dag):
+        task = entrypoint.tasks[0]
+    else:
+        task = entrypoint
+    task_resources = task.get_resources()
+    if task_resources:
+        assert len(task_resources) == 1
+        task_resources = list(task_resources)[0]
+        if task_resources.cpu is not None:
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(f'Cannot specify CPU resources for sky exec. '
```

Comment:
```suggestion
                raise ValueError(f'Cannot specify CPU resources for sky.exec(). '
```

Reply:
Actually, that's exactly what the copilot suggested. However, I intentionally changed it into `sky exec` for the sake of  CLI users.